streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala

```diff
@@ -140,6 +141,18 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
     }
     true
   }
+
+  test("registering listeners via spark.streaming.extraListeners") {
+    val conf = new SparkConf().setMaster("local").setAppName("test")
+      .set("spark.streaming.extraListeners", classOf[BatchInfoCollector].getName + "," +
+        classOf[ReceiverInfoCollector].getName)
+    val scc = new StreamingContext(conf, Seconds(1))
+
+    scc.scheduler.listenerBus.listeners.collect {
```

Comment:
Its easier to understand if this check is replaced with `... listeners.exists { _.isInstanceOf[...] } `